src/pepr/operator/index.ts

```diff
@@ -0,0 +1,27 @@
+import { Capability } from "pepr";
+
+import { namespaceFinalizer } from "./controllers/istio/injection";
+import { UDSPackage } from "./crd";
+import "./crd/register";
+import { validator } from "./crd/validator";
+import { Queue } from "./enqueue";
+
+export const operator = new Capability({
+  name: "uds-core-operator",
+  description: "The UDS Operator is responsible for managing the lifecycle of UDS resources",
+});
+
+export const { Store, When } = operator;
+
+const queue = new Queue();
+
+// Watch for changes to the UDSPackage CRD to remove the finalizer
+When(UDSPackage).IsDeleted().Watch(namespaceFinalizer);
```

Comment:
This is not actually a real k8s finalizer (the `namespaceFinalizer` name is confusing), but it is possible to implement one in Pepr without too much effort. If the watcher somehow misses the deletion event (e.g. operator is shut down long enough to miss the event in the history), this `namespaceFinalizer` won't be invoked.

You could make a real finalizer with `.metadata.finalizers`. You would instead watch without a specific create/update/delete phase (no `IsDeleted()`) and instead check for the `.metadata.deletionTimestamp` and the presense of your finializer name in  `.metadata.finalizers`. This way if a watcher restarts and does the initial list the finalizer cleanup will be correctly invoked. The deletion of a UDSPackage would be blocked until the finalizer completes.

Reply:
yeah just a bad name, I had done it as a finalizer and then changed it as we didnt really need a finalizer here